web/src/routes/(user)/shared-links/[[id=id]]/+page.svelte

```diff
@@ -0,0 +1,119 @@
+<script lang="ts">
+  import { goto } from '$app/navigation';
+  import { page } from '$app/state';
+  import GroupTab from '$lib/components/elements/group-tab.svelte';
+  import UserPageLayout from '$lib/components/layouts/user-page-layout.svelte';
+  import CreateSharedLinkModal from '$lib/components/shared-components/create-share-link-modal/create-shared-link-modal.svelte';
+  import { dialogController } from '$lib/components/shared-components/dialog/dialog';
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import SharedLinkCard from '$lib/components/sharedlinks-page/shared-link-card.svelte';
+  import { AppRoute } from '$lib/constants';
+  import { handleError } from '$lib/utils/handle-error';
+  import { getAllSharedLinks, removeSharedLink, SharedLinkType, type SharedLinkResponseDto } from '@immich/sdk';
+  import { onMount } from 'svelte';
+  import { t } from 'svelte-i18n';
+  import type { PageData } from './$types';
+
+  type Props = {
+    data: PageData;
+  };
+
+  const { data }: Props = $props();
+
+  let sharedLinks: SharedLinkResponseDto[] = $state([]);
+  let sharedLink = $derived(sharedLinks.find(({ id }) => id === page.params.id));
+
+  const refresh = async () => {
+    sharedLinks = await getAllSharedLinks();
+  };
+
+  onMount(async () => {
+    await refresh();
+  });
+
+  const handleDeleteLink = async (id: string) => {
+    const isConfirmed = await dialogController.show({
+      title: $t('delete_shared_link'),
+      prompt: $t('confirm_delete_shared_link'),
+      confirmText: $t('delete'),
+    });
+
+    if (!isConfirmed) {
+      return;
+    }
+
+    try {
+      await removeSharedLink({ id });
+      notificationController.show({ message: $t('deleted_shared_link'), type: NotificationType.Info });
+      await refresh();
```

Comment:
I think we could already optimistically remove the link from the array if the `removeSharedLink` call was successful, and then after that actually re-fetch the links. That could make it feel a bit more snappy

Reply:
True, I'll leave that for a follow-up though. There's definitely a lot of room for improvement though, this was how it was working previously.